src/MOL_discretization.jl

```diff
@@ -1,22 +1,236 @@
+                    expr = :(u[:,$j])
+                elseif deriv_order == 1
+                    # TODO: approx_order and forward/backward should be 
+                    #       input parameters of each derivative
+                    approx_order = 1
+                    L = UpwindDifference(deriv_order,approx_order,dx[i],len[i]-2,-1)
+                    expr = :(-1*($L*Q[$j]*u[:,$j]))
+                elseif deriv_order == 2
+                    L = CenteredDifference(deriv_order,approx_order,dx[i],len[i]-2)
+                    expr = :($L*Q[$j]*u[:,$j])
+                end
+            end
+            return expr
+        elseif input.op isa Differential
+            var = input.op.x.op.name
+            push!(deriv_var,var)
+            return discretize_2(input.args[1],deriv_order+1,approx_order,dx,X,
+                                len,deriv_var,dep_var_idx,indep_var_idx)
+            pop!(deriv_var,var)
+        else
+            if size(input.args,1) == 1
+                aux = discretize_2(input.args[1],deriv_order,approx_order,dx,X,
+                                   len,deriv_var,dep_var_idx,indep_var_idx)
+                return :(broadcast($(input.op), $aux))
+            else
+                aux_1 = discretize_2(input.args[1],deriv_order,approx_order,dx,X,
+                                     len,deriv_var,dep_var_idx,indep_var_idx)
+                aux_2 = discretize_2(input.args[2],deriv_order,approx_order,dx,X,
+                                     len,deriv_var,dep_var_idx,indep_var_idx)
+                return :(broadcast($(input.op), $aux_1, $aux_2))
+            end    
+        end
+    end
+end
+
+# Convert a PDE problem into an ODE problem
 function DiffEqBase.discretize(pdesys::PDESystem,discretization::MOLFiniteDifference)
-  tdomain = pdesys.domain[1].domain
-  domain = pdesys.domain[2].domain
-  @assert domain isa IntervalDomain
-  len = domain.upper - domain.lower
-  dx = discretization.dxs[1]
-  interior = domain.lower+dx:dx:domain.upper-dx
-  X = domain.lower:dx:domain.upper
-  L = CenteredDifference(2,2,dx,Int(len/dx)-2)
-  Q = DirichletBC(0.0,0.0)
-  function f(du,u,p,t)
-    mul!(du,L,Array(Q*u))
-  end
-  u0 = @. - interior * (interior - 1) * sin(interior)
-  PDEProblem(ODEProblem(f,u0,(tdomain.lower,tdomain.upper),nothing),Q,X)
+
+    # TODO: discretize the following cases
+    #
+    #   1) PDE System
+    #        1.a) Transient
+    #                There is more than one indep. variable, including  't'
+    #                E.g. du/dt = d2u/dx2 + d2u/dy2 + f(t,x,y)
+    #        1.b) Stationary
+    #                There is more than one indep. variable, 't' is not included
+    #                E.g. 0 = d2u/dx2 + d2u/dy2 + f(x,y)
+    #   2) ODE System
+    #        't' is the only independent variable
+    #        The ODESystem is packed inside a PDESystem
+    #        E.g. du/dt = f(t)
+    #
+    #   Note: regarding input format, lhs must be "du/dt" or "0".

```

Comment:
Note that it shouldn't always be an ODEProblem. If there isn't a "preferred dimension", then it should become a NonlinearProblem. We should discuss this a bit more.